packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -660,22 +660,47 @@ export class UserPool extends UserPoolBase {
    * Import an existing user pool based on its id.
    */
   public static fromUserPoolId(scope: Construct, id: string, userPoolId: string): IUserPool {
-    class Import extends UserPoolBase {
-      public readonly userPoolId = userPoolId;
-      public readonly userPoolArn = Stack.of(this).formatArn({
-        service: 'cognito-idp',
-        resource: 'userpool',
-        resourceName: userPoolId,
-      });
-    }
-    return new Import(scope, id);
+    let userPoolArn = Stack.of(scope).formatArn({
+      service: 'cognito-idp',
+      resource: 'userpool',
+      resourceName: userPoolId,
+    });
+
+    return UserPool.fromUserPoolArn(scope, id, userPoolArn);
   }
 
   /**
    * Import an existing user pool based on its ARN.
    */
   public static fromUserPoolArn(scope: Construct, id: string, userPoolArn: string): IUserPool {
-    return UserPool.fromUserPoolId(scope, id, Stack.of(scope).parseArn(userPoolArn).resourceName!);
+    let userPoolId = Stack.of(scope).parseArn(userPoolArn).resourceName;
+
+    if (!userPoolId) {
+      throw new Error('invalid user pool ARN');
+    }
+
+    class ImportedUserPool extends UserPoolBase {
+      public readonly userPoolArn: string;
+      public readonly userPoolId: string;
+      constructor(_scope: Construct, _id: string, attrs: {
+        userPoolId: string;
+        userPoolArn: string
+      }) {
+        const arnParts = Stack.of(_scope).parseArn(attrs.userPoolArn);
+        super(_scope, _id, {
+          account: arnParts.account,
+          region: arnParts.region,
+        });
+
+        this.userPoolArn = attrs.userPoolArn;
+        this.userPoolId = attrs.userPoolId;
+      }
+    }
+
+    return new ImportedUserPool(scope, id, {
+      userPoolId: userPoolId,
+      userPoolArn: userPoolArn,
+    });
```

Comment:
Is this simpler? Fewer lines of code to achieve the same thing and removes the need for `_scope` and `_id`.

```suggestion
    class ImportedUserPool extends UserPoolBase {
      public readonly userPoolArn = userPoolArn;
      public readonly userPoolId = userPoolId;
      constructor() {
        const arnParts = Stack.of(scope).parseArn(userPoolArn);
        super(scope, id, {
          account: arnParts.account,
          region: arnParts.region,
        });
      }
    }

    return new ImportedUserPool();
```